incubator/hnc/pkg/controllers/config_controller.go

```diff
@@ -42,13 +56,24 @@ func (r *ConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	// achieve the reconciliation.
 	r.Log.Info("Reconciling cluster-wide HNC configuration.")
 
+	// TODO: This is only for debuging and testing in the early development stage. Remove this when the
+	// implementation is completed.
+	r.logSpec(inst)
+
+	// Create cooresponding ObjectReconcilers for newly added types, if needed.
+	// TODO: Returning an error here will make controller-runtime to requeue the object to be reconciled again.
+	// A better way to handle this is to display the error as part of the HNCConfig.status field to suggest the error is a
+	// permanent problem and to stop controller-runtime from retrying.
+	if err := r.createObjectReconcilers(inst); err != nil {
+		r.Log.Error(err, "Couldn't create the new object reconciler.")
+		return ctrl.Result{}, err
+	}
+
 	// Write back to the apiserver.
 	if err := r.writeSingleton(ctx, inst); err != nil {
```

Comment:
Just curious: what is the scenario that we need to write the singleton during `Reconcile()`?

Reply:
IIUC, if we update HNCConfiguration.Status during reconciliation, we need to write the updated HNCConfiguration object back to the apiserver. Feel free to correct me if this is not the case.

Reply:
This is correct - we're not doing it yet, but you should probably add that as another step in your plan doc.

Reply:
Thanks for the confirmation. Doen.